contribute/design/1.x/apis/cstorvolume_policy.md

```diff
@@ -0,0 +1,344 @@
+# Move `CStorVolumeConfig`, `CStorVolumePolicy` resources Schema to version `v1`
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [Non-Goals](#non-goals)
+* [VolumeConfig Schema Proposal](#volume-config-schema-proposal)
+* [VolumePolicy Schema Proposal](#volume-policy-schema-proposal)
+* [Risk and Mitigations](#risks-and-mitigations)
+* [CRD Migration and Graduation](#graduation-criteria)
+
+## Introduction
+
+This proposal highlights the limitations of current `CStorVolumeConfig` and `CStorVolumePolicy` schema and
+proposes improvements and feature changes.
+
+# Goals
+
+The major goal of this document is to freeze the APIs schema for `CStorVolumeConfig` and `CStorVolumePolicy`.
+Apart from this the document focuses on following aspects too:
+
+- Migrating the `CStorVolumeConfig` and `CStorVolumePolicy` CRD to `v1` apiversion inder `cstor.openebs.io` api group.
+- Identify the breaking and upgrade changes and documentation around that.
+- List of possible high level status fields on `CStorVolumeConfig` and `CStorVolumePolicy`.
+
+# Non-Goals
+
+- Introduction of new fields in the `CStorVolumeConfig` and `CStorVolumePolicy` to deliver feature etc.
+- List of possible events that should be recorded on `CStorVolumeConfig` and `CStorVolumesPolicy`.
+- List of conditions that should be put on the `CStorVolumeConfig` and `CStorVolumePolicy` CRs.
+
+# Volume Config Schema Proposal
+
+## `CStorVolumeConfig` APIs
+
+```go
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// CStorVolumeConfig describes a cstor volume claim resource created as
+// custom resource. CStorVolumeConfig is a request for creating cstor volume
+// related resources like deployment, svc etc.
+type CStorVolumeConfig struct {
```

Comment:
@vishnuitta  @prateekpandey14: Can this be completely eliminated and merged into cStorVolume itself?

Reply:
`CStorVolumeConfig` have these own importence to support different day 2 ops based operations and validations which i have added in PR as well.